### ISSUE TITLE
How to handle '=' in values, splitting on | but KV takes over all '=' not only the first

### ISSUE THREAD

**Issue by Mojster** [logstash-plugins/logstash-filter-kv#51]
I'm parsing custom logs. Here's a snippet:
18.7.2012 9:05:57\t|C3|date=18.07.2012 09:05:57|acronym=BS|... |firstsignUpDate=30.07.2007|bibl001c=m|biblUDK675s=(038)33=111=163.6|....

My second (first one ist just mutate/gsub just to change C3 => cir=C3) filter applied is KV
```
kv {
	field_split => "|"
}
```
This works fine, until my field containts multiples '=', for example `biblUDK675s=(038)33=111=163.6`.
I've thought that after splitting with '|' only the part before first = should be taken as key.
Is there any option to tell KV that `biblUDK675s` is the key and `(038)33=111=163.6` is the value?

LS is currently on 5.5.1 version and running on Windows server 2012 R2 x64

I'm writing here, because didn't got any reply on logstash discus forums. And this issue is for me in top priority.

**Comment by Mojster** [comment]
Found out, that this is also mentioned in #12 .

**Comment by Mojster** [comment]
This one is solved, if I apply:
```
include_brackets => false
```

But if I use a diffrent string:
18.7.2012 9:05:57\t|C3|date=18.07.2012 09:05:57|acronym=BS|… |firstsignUpDate=30.07.2007|bibl001c=m|biblUDK675s=test AU =fgdgd|…

Then it returns:
biblUDK675s=test
AU=fgdgd

But I explicitly applied that field split is "|", why does he uses a fall back?
In first example, after including brackets, it returned correct, because there was no space.

**Comment by Mojster** [comment]
I've abandoned KV filter and wrote a new one for me in ruby code.

If it helps somebody:
```
ruby {
		code => "
			a = event.get('message').split('|').delete_if{|x| !x.match(/=/)}
			a.each {|y| b = y.split('=', 2)
				event.set(b[0].strip, b[1])
			}"
	}
```

**Comment by elvarb** [comment]
Similar problem as I'm having

When I have the KV filter run against this string

`last="3367" post="\" ltime=\"3"`

Instead of ending up with

```
last: 3367
post: " ltime="3
```
I end up with

```
last: 3367
post: \
ltime: \"3"
```

I am expecting the KV filter to accept \ as an escape characters so that the value is encapsulated by the non escaped quotation marks but it seems that the kv seperator process kicks in before the kv value is read completely.

**Comment by Mojster** [comment]
This is normal behaviour like in Java (String.split(" ")) or PHP (explode(" ", String)).

So you split your string by " " and after that in this parts you split it again on "=" to determine the key and the value part.

This is how I understand this, the developer of KV can have other views on this.

**Comment by Mojster** [comment]
I was wrong with the part of trimming the " away.
Because `include_brackets` only trim away brackets ()[]{}...

**Comment by Mojster** [comment]
Closing, because it was fixed in #55
